src/ContainerFactory.php

```diff
@@ -12,14 +12,12 @@
 
 use Pimple\Container;
 use Pimple\Psr11\Container as PsrContainer;
+use Psr\Container\ContainerInterface;
 
 class ContainerFactory
 {
-    public function __invoke(ConfigInterface $config) : PsrContainer
+    public function __invoke(ConfigInterface $config) : ContainerInterface
```

Comment:
BC Break also here. But I believe it can be reverted, as it actually return `PsrContainer` anyway.

Reply:
@michalbundyra I add troubles writing test cases as Pimpl\Psr11\Container is `final` kind regards.

Reply:
@michalbundyra Since at the end all we want to have a `Psr\Container\ContainerInterface` from a pimple-container, even if BC, the `ContainerInterface` return value type-hint may also make more sense. We could swap the pimple-psr11-wrapper with another wrapper (of course this made sense when there was no official wrapper) without changing the signature again. kind regards.

Reply:
in general, imo, we should set our factories to final, and then it wouldn't be BC break... 

In general I think the changes you are proposing here are good, but it requires new major.
Then we should do more updates for the next major - like final factories, maybe marking some classes as internal, so then we can easily update signatures if needed in future.

Reply:
...and I admit this is a bit of a "fussy" pr (...I'm not sure the italian word I mean is correctly translated by this term...) :-)